plugins/microservices/src/json_parse.cpp

```diff
@@ -0,0 +1,130 @@
+/// \file
+
+#include "msi_preconditions.hpp"
+
+#include "irods/irods_error.hpp"
+#include "irods/irods_logger.hpp"
+#include "irods/irods_ms_plugin.hpp"
+#include "irods/irods_re_structs.hpp"
+#include "irods/msParam.h"
+#include "irods/objInfo.h"
+#include "irods/process_stash.hpp"
+#include "irods/rcMisc.h"
+#include "irods/rodsErrorTable.h"
+#include "irods/rodsPackInstruct.h"
+
+#include <nlohmann/json.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <span>
+#include <string>
+
+namespace
+{
+    // clang-format off
+    using log_msi = irods::experimental::log::microservice;
+    using json    = nlohmann::json;
+    // clang-format on
+
+    // NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
+    auto msi_impl(MsParam* _buffer, MsParam* _buffer_size, MsParam* _json_handle, ruleExecInfo_t* _rei) -> int
+    {
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer_size);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_json_handle);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_rei);
+
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer->type);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer_size->type);
+
+        IRODS_MSI_REQUIRE_TYPE(_buffer->type, STR_MS_T, IntArray_MS_T);
+        IRODS_MSI_REQUIRE_TYPE(_buffer_size->type, INT_MS_T);
+
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer->inOutStruct);
+        IRODS_MSI_REQUIRE_VALID_POINTER(_buffer_size->inOutStruct);
+
+        try {
+            const auto buf_size = *static_cast<int*>(_buffer_size->inOutStruct);
+
+            if (buf_size <= 0) {
+                log_msi::error("Invalid argument: Buffer size must be greater than zero.");
+                return SYS_INVALID_INPUT_PARAM;
+            }
+
+            std::string handle;
+
+            if (const std::string_view t = _buffer->type; t == STR_MS_T) {
+                const auto* char_array = static_cast<const char*>(_buffer->inOutStruct);
+                // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+                handle = irods::process_stash::insert(json::parse(char_array, char_array + buf_size));
```

Comment:
Hmm. Perhaps. Depends on what we want to do when an exception is caught. If we're not returning a JSON parse error code, then it may not be worth the extra catch-block.